plugin.rb

```diff
@@ -996,6 +1019,39 @@ def show
       render_serialized query, DataExplorer::QuerySerializer, root: 'query'
     end
 
+    def group_reports_index
+      respond_to do |format|
+        format.html { render 'groups/show' }
+        format.json do        
+          group = Group.find_by(name: params["group_name"])
+          return render(status: 404, json: {}) unless current_user && user_included_in_group(group)
+
+          queries = DataExplorer::Query.all
+          queries.select! { |query| query.group_ids.include?(group.id.to_s) }
+          render_serialized queries, DataExplorer::QuerySerializer, root: 'queries'
+        end
+      end
+    end
+
+    def group_reports_show
+      respond_to do |format|
+        format.html { render 'groups/show' }
+        format.json do    
+          query = DataExplorer::Query.find(params[:id].to_i)
+          return render(status: 404, json: {}) unless current_user && user_can_access_query(query)
+
+          render_serialized query, DataExplorer::QuerySerializer, root: 'query'
+        end
+      end
+    end
+
+    def group_reports_run
+      query = DataExplorer::Query.find(params[:id].to_i)
+      return render(status: 404, json: {}) unless current_user && user_can_access_query(query)
```

Comment:
Our general pattern is to use a guardian eg: that should take care of current_user check and everything. 

https://github.com/discourse/discourse/blob/cd894b7d17a5d0712facb943229cdb2504818a7f/app/controllers/topics_controller.rb#L49-L49